### ISSUE TITLE
to many recipients warning though there is only 1 recipient when replying

### ISSUE THREAD

**Issue by tobyps** [revad/use_bcc_instead_C#8]
Hi,
with TB68.1.1 the "to many recipients" warning ist prompted wrong.

If I create a new message and add recipients all works normal.

If I resply to a message it prompts the warning, though there is only 1 recipient. Deleting the recipient and adding it again solves the problem, no warning ist then prompted.
I tested it with 2 recipients, my limit for the Warning is 10 recipient.

Apart from that error all seems to work OK

**Comment by revad** [comment]
I tried to replicate this - TB 68.1.1 on Linux
I have max number of non-BCC recipients set to 3
All messages plain text.
I sent myself an email (only 1 TO recipient) and replied (only 1 TO recipient.) 
The message was sent.

Maybe your case differs from my test or is more complicated. Can you give further details so I can replicate?

**Comment by tobyps** [comment]
I probably used html messages to test it, but I replicated the test with a plain text message, no difference. My max number is set to 10 but I also tried 8 and 99, no difference.
But I discovered that it only affects my Notebook (Win10), my Desktop (Win 10) doesn't has the issue. 
As far as I could see the only difference were a few AddOns. I deactivated all Addons but it made no difference. Deactivating and Removing Use BCC instead didn't work either.
I checked the settings but there are no differences.
I checked the error console, on both systems a few entries were produced, but none seem to refer to bcc and in both cases the entries were identical.

So far, I have no idea what else I could test or if it even is a problem with this Addon

**Comment by revad** [comment]
So, can you reply to the *same* message, in the same way, on these two machines, and in one the reply is blocked ("Too many non-BCC'ed recipients...") and in the other it is not?

One difference between them (and with my instances of TB) might be the profiles - one carried forward from TB60 and one newly set up, for example. But I can't think why that would affect the addon.

I tried TB70.0b2 and I can't replicate the problem there either.

**Comment by tobyps** [comment]
Yes, I used the same message, Desktop it went out, without problem, Notebook I got the error. 
I sent myself a message, (using two different mailaccounts) and replying one worked and one didn't.

Some underlying issue with the profile would be my guess too. I use the Notebook a lot more and probably tried for example more Addons in the past, but generally both are from before TB60.

**Comment by revad** [comment]
You could try a new profile on the notebook.

Check that the options stored are the same (options, advanced, config editor, filter on 'insteadC')

(There may be options left over from the old 'Use BCC Instead' addon. Ignore those. You can reset them and they'll disappear on reload.)

**Comment by revad** [comment]
On the notebook, have you got a mailing list with the same or similar name to the replied-to recipient? (There's a comment in the code that a mailing list can - or once could - have the same name as an address.)

There is some very old code in the addon, going back to TB v2. Some of it appears to handle address books differently in earlier versions. Maybe the version checks no longer work and this old code is being run. 

Anyway, I have a new version  (it's needed for TB71) which removes this old code. If you want to try it email me at the address on this page
https://addons.mozilla.org/en-US/firefox/addon/mudcat-browser-tools/

**Comment by revad** [comment]
Was the email to which you are replying sent to you with TO, CC, or BCC? (Though I think you would have mentioned if it was BCC.)

Are you using 'Reply to all' or 'Reply to sender'?  (There are extra checks in the 'all' case.)

Are any of your mailbox files near to or over 4GB? (There seems to be some code to address that.)

There are quite a few workrounds which have been included in the code over the years!

**Comment by tobyps** [comment]
TO or CC but the tests I did were all TO.

Always reply to sender. I will test if it makes a difference using reply to all. 

All Mailboxes are well below 4 GB but  I always get the message if I want to compress the folder because they are to big (the message says 2400 GB and it doesn't help to do it). As I don't restart TB often I never tried to solve this issue.

I appreciate your help very much, but as it works without problems with the new profile, you don't need to try to fix the issue. Though I'm happy to further test everything you have, if you want to continue.

**Comment by revad** [comment]
OK. We'll blame this on an anomaly in the profile.

I'll leave it open for a while in case others have the same problem.
